tests/fixtures/cloud_uploader.py

```diff
@@ -0,0 +1,102 @@
+import json
+import os
+import subprocess
+import typing
+
+
+class Uploader(object):
+    def __init__(self, local_root: str) -> None:
+        self.local_root = local_root
+
+    def reset(self) -> None:
+        raise NotImplementedError()
+
+    def upload_file(self, local_path: str, remote_path: str, metadata_keys: typing.Dict[str, str]=None) -> None:
+        raise NotImplementedError()
+
+
+class S3Uploader(Uploader):
+    def __init__(self, local_root: str, bucket: str) -> None:
+        super(S3Uploader, self).__init__(local_root)
+        self.bucket = bucket
+
+    def reset(self) -> None:
+        args = [
+            "aws",
+            "s3",
+            "rm",
+            "s3://{}".format(self.bucket),
+            "--recursive"
+        ]
+        subprocess.check_call(args)
+
+    def upload_file(
+            self,
+            local_path: str,
+            remote_path: str,
+            metadata_keys: typing.Dict[str, str]=None,
+            tags: typing.Dict[str, str]=None) -> None:
+        if metadata_keys is None:
+            metadata_keys = dict()
+        if tags is None:
+            tags = dict()
+        args = [
+            "aws", "s3", "cp",
+            os.path.join(self.local_root, local_path),
+            "s3://{}/{}".format(self.bucket, remote_path),
+            "--metadata",
+            json.dumps(metadata_keys)
+        ]
+        subprocess.check_call(args)
```

Comment:
Why not use `boto3.resource("s3").upload_file()` and the corresponding boto3 method for tagging instead of spawning a subprocess? It calls the same code on the backend.